File: packages/ts-sdk/test/integration/live-api.test.ts
import { describe, expect, it } from 'vitest';
import { TreeDbClient } from '../../src/treedb/index.js';

describe('live TreeDB API integration', () => {
  it('runs health when TREEDB_BASE_URL is configured or reports not configured cleanly', async () => {
    const baseUrl = process.env.TREEDB_BASE_URL;
    if (!baseUrl) {
      expect({ status: 'not_configured', reason: 'TREEDB_BASE_URL is not set' }).toMatchObject({ status: 'not_configured' });
      return;
    }

    const client = new TreeDbClient({ baseUrl, token: process.env.TREEDB_TOKEN });
    await expect(client.health()).resolves.toBeDefined();
  });
});

File: packages/elixir-sdk/test/integration/live_api_test.exs
defmodule TreeDbSdk.LiveApiTest do
  use ExUnit.Case, async: true

  test "live health is optional" do
    case System.get_env("TREEDB_BASE_URL") do
      nil ->
        assert true

      base_url ->
        client = TreeDbSdk.Client.new(base_url: base_url, token: System.get_env("TREEDB_TOKEN"))
        assert {:ok, _} = TreeDbSdk.health(client)
    end
  end
end
